### PR TITLE
Support opening password protected PDF files

### LINKED ISSUE #712: Support Password protected PDF file
It would be good to be able to open encrypted PDF file.

Currently APP just return toast `Error:java.lang.SecurityException: password required...` when opening this kind of files

### PR DESCRIPTION
**Pull request by esensar**
This closes #712

## TODO
- [x] Merge https://github.com/SimpleMobileTools/Simple-Commons/pull/1725 and update ref in `build.gradle`

### CODE REVIEW

File: app/src/main/res/values-ar/strings.xml [hunk 1]
```diff
@@ -50,11 +50,15 @@
     <string name="others">الاخرين</string>
     <string name="storage_free">free</string>
     <string name="total_storage">إجمالي مساحة التخزين: %s</string>
+    <!-- Password protected files -->
+    <string name="enter_password">Enter password</string>
+    <string name="password">Password</string>
+    <string name="empty_password">Please enter the password</string>
```
Comment (tibbi) [inline app/src/main/res/values-ar/strings.xml:56]: what error is shown at wrong password?

Comment (esensar) [reply app/src/main/res/values-ar/strings.xml:56]: It is currently displayed as a toast using the exception message (the code which was already there). For english it says: `Error: Password required or incorrect password`

Do you want me to add a custom message for that case too?

Comment (tibbi) [reply app/src/main/res/values-ar/strings.xml:56]: lets actually move these strings into Commons at  section, they could be reused in other apps too

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/PDFViewerActivity.kt [hunk 2]
```diff
@@ -91,15 +97,34 @@ class PDFViewerActivity : SimpleActivity() {
...
         pdf_viewer.fromUri(uri)
+            .password(filePassword)
             .scrollHandle(DefaultScrollHandle(this, primaryColor.getContrastColor(), primaryColor))
             .spacing(15)
             .onTap { toggleFullScreen() }
             .onError {
-                showErrorToast(it.localizedMessage?.toString() ?: getString(R.string.unknown_error_occurred))
-                finish()
+                if (it is PdfPasswordException) {
+                    // Already entered a password and it was wrong
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/PDFViewerActivity.kt:113]: lets start comments with lowercase for consistency

File: app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/PDFViewerActivity.kt [hunk 3]
```diff
@@ -91,15 +97,34 @@ class PDFViewerActivity : SimpleActivity() {
...
+                    if (filePassword != null) {
+                        showErrorToast(getString(R.string.invalid_password))
+                        finish()
+                    } else {
+                        EnterPasswordDialog(
+                            this,
+                            callback = { password -> loadPdfViewer(uri, password) },
+                            cancelCallback = { finish() }
```
Comment (tibbi) [inline app/src/main/kotlin/com/simplemobiletools/filemanager/pro/activities/PDFViewerActivity.kt:121]: split these into multiple lines please

**Comment by tibbi** [comment]
do not close the whole Enter password dialog after inserting an invalid password, just show a toast and clear the password field

**Comment by tibbi** [comment]
looks good now, thanks

**Comment by typebrook** [comment]
Many thanks! @esensar @tibbi

**Pull request merged**